### PR TITLE
[SLP] Enable optimization of freeze instructions

### PR DESCRIPTION
**Pull request by jmciver**
Allow SLP optimization to progress in the presence of freeze instructions. Prior
to this commit, freeze instructions blocked SLP optimization.
    
The following URL shows correctness of the addsub_freeze test:
https://alive2.llvm.org/ce/z/qm38oh

### CODE REVIEW

File: llvm/lib/Transforms/Vectorize/SLPVectorizer.cpp [hunk 1]
```diff
@@ -9790,10 +9792,11 @@ BoUpSLP::getEntryCost(const TreeEntry *E, ArrayRef<Value *> VectorizedVals,
   case Instruction::AShr:
   case Instruction::And:
   case Instruction::Or:
-  case Instruction::Xor: {
+  case Instruction::Xor:
+  case Instruction::Freeze: {
```
Comment (alexey-bataev) [inline llvm/lib/Transforms/Vectorize/SLPVectorizer.cpp:9796]: I rather doubt it should be here. I think the cost of freeze must be 0 always

Comment (jmciver) [reply llvm/lib/Transforms/Vectorize/SLPVectorizer.cpp:9796]: Thanks for the feedback. You are correct. I will get that fixed.

File: llvm/lib/Transforms/Vectorize/SLPVectorizer.cpp [hunk 2]
```diff
@@ -10118,6 +10120,9 @@ BoUpSLP::getEntryCost(const TreeEntry *E, ArrayRef<Value *> VectorizedVals,
     };
     return GetCostDiff(GetScalarCost, GetVectorCost);
   }
+  case Instruction::Freeze: {
+    return CommonCost;
+  }
```
Comment (alexey-bataev) [inline llvm/lib/Transforms/Vectorize/SLPVectorizer.cpp:10125]: Drop braces

**Review by alexey-bataev — APPROVED** [review]
LG

**Comment by jmciver** [comment]
@alexey-bataev I currently do not have committer access. Can you please land the pull request for me?

Would you like me to squash + push first?

**Pull request merged**